basic-plus.html

```diff
@@ -158,18 +158,23 @@ <h3 class="panel-title">Demo Notes</h3>
             .test(window.navigator.userAgent),
         previewMaxWidth: 100,
         previewMaxHeight: 100,
-        previewCrop: true
+        previewCrop: true,
+        singleFileUploads: false
     }).on('fileuploadadd', function (e, data) {
         data.context = $('<div/>').appendTo('#files');
         $.each(data.files, function (index, file) {
             var node = $('<p/>')
-                    .append($('<span/>').text(file.name));
-            if (!index) {
+                .append($('<span/>').text(file.name));
+            if (data.files.length <= 1) {
```

Comment:
A better check would be the following, which matches both the single file selection case as well as the last item in a list of multiple selections:
```
if (index === data.files.length - 1) {
```
This has the benefit of making the additional section you added obsolete and thus results in a one-line change.

Reply:
Actually I first tried this solution, and I find the layout is weird when set singleFileUploads to false.
The button was embeded in the last picture's parent element p. 
```html
<p>
<canvas/>
</p>
<p>
<canvas/>
</p>
<p>
<canvas/>
<button/>
</p>
```
I am trying to fix this, with my current solution **the button will be sibling of the paragraphs** which store the picture.
```html
<p>
<canvas/>
</p>
<p>
<canvas/>
</p>
<p>
<canvas/>
</p>
<button/>
```

However, if you persist to embed the button to last picture's parent element. I will revert the code back.

Reply:
Please confirm if you want to accept my suggestion or keep it simple with one line changed.

Reply:
I agree with you, adding the button outside of the last element makes more sense.
However you can achieve this by changing it in the first section, basically appending the button to `data.context` instead of the `node`.

Reply:
What did you mean by `first section`?  The node contains p element first
When singleFileUploads  is true, **the button should  append to p element**, 
and when singleFileUploads  is false, the button should append to div element(which is data.context).
I did not find a way to make the code more elegant.
```JavaScript
on('fileuploadadd', function (e, data) {
        data.context = $('<div/>').appendTo('#files');
        $.each(data.files, function (index, file) {
            var node = $('<p/>')
                    .append($('<span/>').text(file.name));
            if (data.files.length <= 1) {
                node
                    .append('<br>')
                    .append(uploadButton.clone(true).data(data));
            }
            node.appendTo(data.context);
            if (data.files.length > 1 && index === data.files.length - 1) {
                var button = uploadButton.clone(true).data(data);
                button.appendTo(data.context);
            }
        });
    })
```

Reply:
By the way, did you close the issue tab of this repository?

Reply:
Yes, the issues are closed, since there were too many users posting requests for support, which I do not provide here anymore.

I think the most elegant and non-disruptive change to this demo code is changing the `fileuploadadd` callback to the following:

```js
function (e, data) {
        data.context = $('<div/>').appendTo('#files');
        $.each(data.files, function (index, file) {
            $('<p/>')
                .append($('<span/>').text(file.name))
                .appendTo(data.context);
            if (index === data.files.length - 1) {
                $('<p/>')
                    .append(uploadButton.clone(true).data(data))
                    .appendTo(data.context);
            }
        });
    }
```

This adds the upload button as sibling at the end of the individual file paragraphs.
It will keep an empty `<p>` around when the upload completes, but before it also kept an empty `<br>` around.

If you change the code to this function, I'll gladly merge it. :)

Reply:
OK, you put the button into another `<p>` element,  and make the `<p>`  as sibling of the container of picture. It's more elegant now. I have pushed a new commit, please verify it.